### PR TITLE
UAR-342: Review trust information interrupt

### PR DESCRIPTION
**Pull request by joshuahigginsonCH**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-342

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/utils/trust.spec.ts [hunk 1]
```diff
@@ -562,6 +563,44 @@ describe('Trust Utils method tests', () => {
...
+    });
+    test("test checkEntityRequiresManageTrusts with application data with no trusts to review", () => {
+      const appData: ApplicationData = {
+        update: {
+          review_trusts: []
+        }
+      };
+      const result = checkEntityRequiresManageTrusts(appData);
+      expect(result).toEqual(false);
+    });
```
Comment (corybudd-ch) [inline test/utils/trust.spec.ts:601]: small but could you add a unit test to cover the review_trusts key doesn't exist in update as well just to make sure that's covered

Comment (corybudd-ch) [reply test/utils/trust.spec.ts:601]: actually missed your first test there covers that

Comment (corybudd-ch) [reply test/utils/trust.spec.ts:601]: what about a test for appdata = undefined? Maybe that's what Sonar is saying is uncovered

Comment (joshuahigginsonCH) [reply test/utils/trust.spec.ts:605]: fixed

File: src/utils/trusts.ts [hunk 2]
```diff
@@ -30,6 +30,13 @@ const checkEntityRequiresTrusts = (appData: ApplicationData | undefined): boolea
     containsTrusteeNatureOfControl(appData.beneficial_owners_corporate ?? []);
 };
 
+const checkEntityRequiresManageTrusts = (appData: ApplicationData | undefined): boolean => {
```
Comment (AndreKramer7654) [inline src/utils/trusts.ts:33]: return appData?.update?.review_trusts && appData.update.review_trusts.length > 0;
- if we wanted to do it all on one line

Comment (AndreKramer7654) [reply src/utils/trusts.ts:33]: Presumably we will have loaded trusts earlier. Only if the manage trusts flag is true.

File: src/controllers/update/beneficial.owner.type.controller.ts [hunk 3]
```diff
@@ -70,7 +70,7 @@ export const postSubmit = (req: Request, res: Response) => {
 
   // Currently only based on flag being on. Will eventually be
```
Comment (AndreKramer7654) [inline src/controllers/update/beneficial.owner.type.controller.ts:71]: delete 71 and 72 comment lines now?

File: src/utils/trusts.ts (hunk 2 above)
Comment (joshuahigginsonCH) [reply src/utils/trusts.ts:33]: this complains about possibly undefined so current way came out as best solution

File: src/controllers/update/beneficial.owner.type.controller.ts (hunk 3 above)
Comment (joshuahigginsonCH) [reply src/controllers/update/beneficial.owner.type.controller.ts]: removed

**Review by cdrainCH — APPROVED** [review]

**Review by AndreKramer7654 — APPROVED** [review]

File: test/utils/trust.spec.ts [hunk 4]
```diff
@@ -562,6 +563,48 @@ describe('Trust Utils method tests', () => {
...
+
+    test("test checkEntityRequiresManageTrusts with undefined application data with no review trusts to review", () => {
+      const result = checkEntityRequiresManageTrusts(undefined);
+      expect(result).toEqual(false);
+    });
+
+    test("test checkEntityRequiresManageTrusts with application data with trusts to review", () => {
+      const appData: ApplicationData = {
```
Comment (AndreKramer7654) [inline test/utils/trust.spec.ts:579]: maybe pull out into a MOCK object

**Pull request merged**